pil-analyzer/src/expression_processor.rs

```diff
@@ -189,7 +190,34 @@ impl<'a, D: AnalysisDriver> ExpressionProcessor<'a, D> {
                 self.process_block_expression(statements, expr, src)
             }
             PExpression::FreeInput(_, _) => panic!(),
-            PExpression::StructExpression(_, _) => unimplemented!("Structs are not supported yet"),
+            PExpression::StructExpression(
+                src,
+                StructExpression {
+                    name: reference,
+                    fields,
+                },
+            ) => {
+                let type_args = reference
+                    .type_args
+                    .map(|args| args.into_iter().map(|t| self.process_type(t)).collect());
+
+                Expression::StructExpression(
+                    src,
+                    StructExpression {
+                        name: Reference::Poly(PolynomialReference {
+                            name: self.driver.resolve_type_ref(&reference.path),
+                            type_args,
+                        }),
+                        fields: fields
```

Comment:
I think we should check here that all the fields exist and that they are also all mentioned.

Reply:
The biggest problem with this is that at this point we don't have the struct declaration. So this check is done during type checking (which was also improved).

Reply:
The AnalysisDriver knows about all the symbols, so we could do it at this point. Not saying we should if it's too difficult.